src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -88,18 +114,6 @@ public VisualStudioProject CreateAndAddToWorkspace(string projectSystemName, str
                 // If we don't have any projects and this is our first project being added, then we'll create a new SolutionId
                 if (w.CurrentSolution.ProjectIds.Count == 0)
                 {
-                    // Fetch the current solution path. Since we're on the UI thread right now, we can do that.
-                    string? solutionFilePath = null;
-                    var solution = (IVsSolution)Shell.ServiceProvider.GlobalProvider.GetService(typeof(SVsSolution));
-                    if (solution != null)
-                    {
-                        if (ErrorHandler.Failed(solution.GetSolutionInfo(out _, out solutionFilePath, out _)))
-                        {
-                            // Paranoia: if the call failed, we definitely don't want to use any stuff that was set
-                            solutionFilePath = null;
-                        }
-                    }
-
                     var solutionSessionId = GetSolutionSessionId();
```

Comment:
Is this now running off the UI thread? Is it safe to do so?

Reply:
yes.  we do basic stuff on immutable data, and the one call we make is to an external object (TelemetrySession) doc'ed as:

> All methods are thread-safe and return immediately.